src/java.desktop/windows/native/libawt/windows/Devices.cpp

```diff
@@ -95,7 +95,22 @@ HMONITOR* g_hmpMonitors;
 // Callback for CountMonitors below
 BOOL WINAPI clb_fCountMonitors(HMONITOR hMon, HDC hDC, LPRECT rRect, LPARAM lP)
 {
-    g_nMonitorCounter ++;
+    // Ignore monitors where CreateDC fails
+    MONITORINFOEX mieInfo;
+    memset((void*)(&mieInfo), 0, sizeof(MONITORINFOEX));
+    mieInfo.cbSize = sizeof(MONITORINFOEX);
+    if (TRUE == ::GetMonitorInfo(hMon, (LPMONITORINFOEX)(&mieInfo))) {
```

Comment:
The documentation for [`GetMonitorInfo`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmonitorinfow) says, _<q cite="https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmonitorinfow#return-value">If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.</q>_

Whereas `TRUE == 1`; therefore the condition should be != 0 or rather simply `if (::GetMonitorInfo(/*params*/))`.

Reply:
Right, that's what's written. But (in the documentation,) the function is declared as `BOOL GetMonitorInfoW...`. So I could epxect retrun values of TRUE/FALSE, no?

Reply:
Addressed.

Reply:
> So I could epxect return values of TRUE/FALSE, no?

Likely, yes… But `BOOL` is `int` in disguise and can have other values…